### PR TITLE
DEV-3658: Add cider and teal to pipeline

### PR DESCRIPTION
**Pull request by hongwingl**

### CODE REVIEW

**Comment by hongwingl** [comment]
I have ran the smoke test and also COLO mini. Still trying to figure out how to run pipeline5 on a real sample.

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    @Override
+    public List<BashCommand> tumorReferenceCommands(final SomaticRunMetadata metadata) {
+        return tumorOnlyCommands(metadata);
+    }
+
+    @Override
+    public List<BashCommand> tumorOnlyCommands(final SomaticRunMetadata metadata) {
+        List<String> arguments = ciderArguments(metadata);
+        // only need 8G for DNA samples. RNA samples would need 32G
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java:63]: This pipeline will never run on RNA so we can probably remove this comment

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript startupScript, final ResultsDirectory resultsDirectory) {
+        return ImmutableVirtualMachineJobDefinition.builder()
+                .name("cider")
+                .startupCommand(startupScript)
+                .performanceProfile(custom(HmfTool.CIDER.getCpus(), HmfTool.CIDER.getMemoryGb()))
+                .namespacedResults(resultsDirectory)
+                .build();
+    }
+
+    // <sampleId>.cider.layout.gz
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java:91]: unclear what these comments mean, generally we don't leave comments in this code.

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    // <sampleId>.cider.locus_stats.tsv
+    // <sampleId>.cider.vdj.tsv.gz
+    // <sampleId>.cider.bam
+    @Override
+    public CiderOutput output(final SomaticRunMetadata metadata, final PipelineStatus jobStatus, final RuntimeBucket bucket,
+            final ResultsDirectory resultsDirectory) {
+        ImmutableCiderOutputLocations.Builder outputLocationsBuilder = CiderOutputLocations.builder();
+        //.outputDirectory(GoogleStorageLocation.of(bucket.name(), resultsDirectory.path(), true));
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/Cider.java:99]: remove commented code

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/CiderOutput.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.hartwig.pipeline.tertiary.cider;
+
+import java.util.Optional;
+
+import com.hartwig.pipeline.StageOutput;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface CiderOutput extends StageOutput {
```
Comment (pauldwolfe) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/CiderOutput.java:10]: Will Cider or Teal output ever be used by a downstream stage? If not we can remove a lot complexity by making these output classes only have the name and no output location

Comment (hongwingl) [reply cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/CiderOutput.java:10]: Orange report might need them later. Would this mean I should keep them?

**Review by pauldwolfe — APPROVED** [review]

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/CiderOutput.java (hunk 4 above)
Comment (pauldwolfe) [reply cluster/src/main/java/com/hartwig/pipeline/tertiary/cider/CiderOutput.java:10]: Yup, sounds good.

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/teal/Teal.java [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+                .namespacedResults(resultsDirectory)
+                .performanceProfile(custom(HmfTool.TEAL.getCpus(), HmfTool.TEAL.getMemoryGb()))
+                .build();
+    }
+
+    // <refSampleId>.teal.tellength.tsv
+    // <refSampleId>.teal.telbam.bam
+    // <tumorSampleId>.teal.tellength.tsv
+    // <tumorSampleId>.teal.telbam.bam
+    // <tumorSampleId>.teal.breakend.tsv.gz
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/teal/Teal.java:164]: Remove commented code

File: cluster/src/main/java/com/hartwig/pipeline/tertiary/teal/Teal.java [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> inputs() {
+
+        // add all download of the super class
```
Comment (nedleitch) [inline cluster/src/main/java/com/hartwig/pipeline/tertiary/teal/Teal.java:80]: Remove commented code

**Review by nedleitch — CHANGES_REQUESTED** [review]

**Pull request merged**